learning-environment/sessions.js

```diff
@@ -386,22 +382,28 @@ var initSessions = function() {
     });
     
     socket.on('query fetch response', function(query_result) {
-            // first clear existing labels
-            clearResultLabels();
-            var max = 1;
-            //calc max
-            for (let i = 0; i < query_result.length; i++) {
-              let val = parseInt(query_result[i][1]);
-              if(val > max) max = val;
-            }
-
-            for (let i = 0; i < query_result.length; i++) {
-                let pos = query_result[i][0];
-                let p = {x: pos[0], y: pos[1], z: pos[2]};
-                let val = parseInt(query_result[i][1]);                
-                let l = glviewer.addLabel(val, {position : p,backgroundOpacity:0.5+(val/max)*.5});                
-                result_labels.push(l);                
-            }
-    });
+        
+        // First clear existing labels
+        clearResultLabels();
+        
+        let max = 1;
+        let i = 0;
+      
+        // Calculate max
+        while (i < query_result.length) {
```

Comment:
why was this changed to a while loop? Using a while loop for simple iteration is poor coding practice. If you want a more modern pattern use forEach.

Reply:
@dkoes I tried benching them now, their performance benefit compared to forEach is a bit on the lower end, I will make the change and send it in sometime :)